### DISCUSSION TITLE
See all the asteroides?

### DISCUSSION THREAD

**Discussion by YvanMiniz** [Stellarium/stellarium#3774] [General]
Hi from Spain. I have been trying to check the possibility to see all the asteroides on the map, but I did not manage. I know we can import an asteroid or a list. But having such a list, I want to see them in order to be able to identify some which I would like to observe and take picture on a day and on an hour that I want. Is it possible ? If so, how ?
Many thanks for your support.
Yvan
Clear skyes

**Comment by gzotti** [comment]
I am afraid we have no functionality to "show all asteroids, although their visual magnitude is far too low for the current simulation conditions". Also, you would have to import an insane number of asteroids for a naive "observe whatever is here" approach, which would slow down the program below usability. 

I have been dreaming for years that some contributor would come up with a plugin that 
- loads >200.000 asteroid data on the GPU (fallback: OpenCL?)
- computes everything (position, magnitude, ...) on GPU
- displays an arbitrary number of asteroids, sorted by various criteria:
-- serial number
-- apparent magnitude
-- physical diameter
-- other physical parameters (albedo, ...)
-- asteroid groups (Trojans, Apollos, etc.): Categorize all objects, allow group selection ("only Trojans") and adding colors
- allow selection of several objects by name/number, display their orbits

Even more ambitious, do a numerical integration and take the planets' gravity into account. This would allow to accurately simulate for centuries. But nasty numerical effects (too large steps with fast simulation, ...) may become problematic. 

The outside view of the "rotating doughnut" of particles is stunning!

**Comment by YvanMiniz** [comment]
Thanks for your quick answer.
Take care.
________________________________
De : Georg Zotti ***@***.***>
Envoyé : vendredi 21 juin 2024 10:39
À : Stellarium/stellarium ***@***.***>
Cc : YvanMiniz ***@***.***>; Author ***@***.***>
Objet : Re: [Stellarium/stellarium] See all the asteroides? (Discussion #3774)


I am afraid we have no functionality to "show all asteroids, although their visual magnitude is far too low for the current simulation conditions". Also, you would have to import an insane number of asteroids for a naive "observe whatever is here" approach, which would slow down the program below usability.

I have been dreaming for years that some contributor would come up with a plugin that

  *   loads >200.000 asteroid data on the GPU (fallback: OpenCL?)
  *   computes everything (position, magnitude, ...) on GPU
  *   displays an arbitrary number of asteroids, sorted by various criteria:
-- serial number
-- apparent magnitude
-- physical diameter
-- other physical parameters (albedo, ...)
-- asteroid groups (Trojans, Apollos, etc.): Categorize all objects, allow group selection ("only Trojans") and adding colors
  *   allow selection of several objects by name/number, display their orbits

Even more ambitious, do a numerical integration and take the planets' gravity into account. This would allow to accurately simulate for centuries. But nasty numerical effects (too large steps with fast simulation, ...) may become problematic.

The outside view of the "rotating doughnut" of particles is stunning!
